File: AUGUST 24/12 August/703. Kth Largest Element in a Stream LC.cpp
class KthLargest {
public:
    priority_queue<int> pq;
    int K;
    KthLargest(int k, vector<int>& nums) {
        K = k;
        for(int num:nums)
            pq.push(-1*num);
        while(pq.size()>k)
            pq.pop();
    }
    
    int add(int val) {
        pq.push(-1*val);
        if(pq.size()>K)
            pq.pop();
        return -1*pq.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */

File: APRIL 2025/AMZN/703. Kth Largest Element in a Stream LC.java
class KthLargest {
    private int K;
    private PriorityQueue<Integer> pq = new PriorityQueue<>();
    public KthLargest(int k, int[] nums) {
        this.K = k;
        for(int num:nums)
            pq.add(num);
    }
    
    public int add(int val) {
        pq.add(val);
        
        while(pq.size()>K)
            pq.poll();
        
        return pq.peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */
